express/src/controllers/create-account.ts

```diff
@@ -7,104 +7,73 @@ import {
 import {randomUUID} from "crypto";
 import {NextFunction, Request, Response} from "express";
 import CognitoInterface from "../lib/cognito/CognitoInterface";
+import {RedirectError, SelfServiceErrors} from "../lib/errors";
 import LambdaFacadeInterface from "../lib/lambda-facade/LambdaFacadeInterface";
 import {SelfServiceError} from "../lib/SelfServiceError";
 
 export const showGetEmailForm = function (req: Request, res: Response) {
-    res.render("create-account/get-email.njk", {values: {emailAddress: req.session.emailAddress}});
+    res.render("create-account/get-email.njk");
 };
 
-export const processGetEmailForm = async function (req: Request, res: Response) {
+export const processGetEmailForm = async function (req: Request, res: Response, next: NextFunction) {
     const emailAddress: string = req.body.emailAddress;
     const cognitoClient: CognitoInterface = await req.app.get("cognitoClient");
 
-    req.session.emailAddress = emailAddress;
-    let result: any;
     try {
-        result = await cognitoClient.createUser(emailAddress);
-        console.debug(result);
+        await cognitoClient.createUser(emailAddress);
     } catch (error) {
         if (error instanceof UsernameExistsException) {
-            // Actually, we should check whether they've verified their email
-            // If not, we should redirect them to the verify email page and tell them to check their email
-            // That page should offer the chance to send a new code to their email address
-            res.redirect("/sign-in");
-            return;
+            // TODO We need to handle this properly with another flow
+            throw new RedirectError("/sign-in");
```

Comment:
I am not sure about this, we can just redirect here - but we can sort this pattern out later